crates/runtime/src/query_history.rs

```diff
@@ -0,0 +1,313 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use std::{
+    sync::Arc,
+    time::{Duration, SystemTime},
+};
+
+use crate::component::dataset::acceleration::Acceleration;
+use crate::component::dataset::TimeFormat;
+use arrow::{
+    array::{BooleanArray, RecordBatch, StringArray, TimestampNanosecondArray, UInt64Array},
+    datatypes::{DataType, Field, Schema, TimeUnit},
+};
+use datafusion::sql::TableReference;
+use snafu::{ResultExt, Snafu};
+use tokio::sync::RwLock;
+
+use crate::{
+    accelerated_table::{refresh::Refresh, AcceleratedTable, Retention},
+    datafusion::DataFusion,
+    dataupdate::DataUpdate,
+    internal_table::create_internal_accelerated_table,
+};
+
+pub const DEFAULT_QUERY_HISTORY_TABLE: &str = "query_history";
+
+pub async fn instantiate_query_history_table() -> Result<Arc<AcceleratedTable>, Error> {
+    let time_column = Some("start_time".to_string());
+    let time_format = Some(TimeFormat::UnixSeconds);
+
+    let retention = Retention::new(
+        time_column.clone(),
+        time_format,
+        Some(Duration::from_secs(3 * 24 * 60 * 60)), // 3 days
```

Comment:
Let's make this 1 day by default.
```suggestion
        Some(Duration::from_secs(1 * 24 * 60 * 60)), // 1 day
```